llvm/test/CodeGen/NVPTX/lower-byval-args.ll

```diff
@@ -31,20 +31,16 @@ define dso_local ptx_kernel void @read_only(ptr nocapture noundef writeonly %out
 ; SM_60-SAME: ptr noundef writeonly captures(none) [[OUT:%.*]], ptr noundef readonly byval([[STRUCT_S:%.*]]) align 4 captures(none) [[S:%.*]]) local_unnamed_addr #[[ATTR0:[0-9]+]] {
 ; SM_60-NEXT:  [[ENTRY:.*:]]
 ; SM_60-NEXT:    [[S3:%.*]] = addrspacecast ptr [[S]] to ptr addrspace(101)
-; SM_60-NEXT:    [[OUT1:%.*]] = addrspacecast ptr [[OUT]] to ptr addrspace(1)
-; SM_60-NEXT:    [[OUT2:%.*]] = addrspacecast ptr addrspace(1) [[OUT1]] to ptr
```

Comment:
With explicit ASCs gone from IR, it may be useful to add a llc test here and in other affected tests, so we can verify that we're still correctly inferring AS of the arguments.

Reply:
Sounds good, I've added llc checks to this test, the other 3 all already check PTX. The only differences are minor re-orderings.